File: golang/551.go
func checkRecord(s string) bool {
    a, l := 0, 0
    for i := 0; i < len(s); i++ {
        if s[i] == \u0027L\u0027 {
            l++
        } else {
            l = 0
        }
        if s[i] == \u0027A\u0027 {
            a++
        }
        if (a >= 2 || l >= 3) {
            return false
        }
    }
    return true
}

File: cpp/551.cpp
class Solution {
public:
    bool checkRecord(string s) {
        int a = 0, l = 0;
        for (int i = 0; i < s.size(); ++i) {
            if (s[i] == \u0027L\u0027)
                ++l;
            else
                l = 0;
            if (s[i] == \u0027A\u0027)
                ++a;
            if (a >= 2 || l >= 3)
                return false;
        }
        return true;
    }
};